openssl/src/lib.rs

```diff
@@ -147,6 +147,8 @@ mod bio;
 #[macro_use]
 mod util;
 pub mod aes;
+#[cfg(ossl320)]
+pub mod argon2;
```

Comment:
Should this just be a `kdf` module?

Reply:
I don't have a strong opinion on this. The other KDFs all live in `pkcs5` right now since that's where their headers were declared. We could do this as `kdf`, although then we might want it to be visible for 3.0+ just with the argon2id function missing (since technically EVP_KDF and EVP_KDF_CTX were added in 3.0).